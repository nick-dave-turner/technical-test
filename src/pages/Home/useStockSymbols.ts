import React, { useEffect } from "react";

import { getStockSymbols } from "../../_MOCKS_";

const useStockSymbols = () => {
  const [stockSymbols, updateStockSymbols] = React.useState<
    Array<IStockSymbol>
  >([]);

  useEffect(() => {
    getStockSymbols()
      .then((result: Array<IStockSymbol>) => updateStockSymbols(result))
      .catch(error => console.log("Error retrieving stock symbols"));
  }, []);

  return { stockSymbols };
};

export default useStockSymbols;
