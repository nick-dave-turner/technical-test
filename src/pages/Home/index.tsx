import React from "react";
import styled from "styled-components";

import TradeForm from "../../components/TradeForm/index";
import TradesTable from "../../components/TradesTable";

import useStockSymbols from "./useStockSymbols";

const Grid = styled.div`
  display: grid;
  grid-template-columns: 1fr 4fr;
  grid-template-rows: auto;
  grid-column-gap: 20px;
  grid-row-gap: 20px;
  grid-template-areas: "sidebar tradesTable" "sidebar stocksTable";

  @media (max-width: 768px) {
    grid-template-columns: 1fr;
    grid-template-areas: "sidebar" "tradesTable" "stocksTable";
  }
`;

const SideBar = styled.aside`
  grid-area: sidebar;
`;

const TradesTableContainer = styled.section`
  grid-area: tradesTable;
`;

const TRADES_DATA: Array<ITrade> = [];

const Home: React.FC = () => {
  const { stockSymbols } = useStockSymbols();

  return (
    <Grid>
      <SideBar>
        <TradeForm symbols={stockSymbols} />
      </SideBar>
      <TradesTableContainer>
        <TradesTable rowData={TRADES_DATA} />
      </TradesTableContainer>
    </Grid>
  );
};

export default Home;
