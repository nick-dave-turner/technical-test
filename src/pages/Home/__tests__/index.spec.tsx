import React from "react";
import { shallow } from "enzyme";

import Home from "../";

describe("Home component", () => {
  it("should match snapshot", () => {
    const component = shallow(<Home />);
    expect(component).toMatchSnapshot();
  });
});
