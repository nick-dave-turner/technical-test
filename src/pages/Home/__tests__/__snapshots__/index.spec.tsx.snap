// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match snapshot 1`] = `
<styled.div>
  <styled.aside>
    <TradeForm
      symbols={Array []}
    />
  </styled.aside>
  <styled.section>
    <TradesTable
      rowData={Array []}
    />
  </styled.section>
</styled.div>
`;
